tools/moth.py

```diff
@@ -328,15 +334,35 @@ def __init__(self, path, seed):
         self.path = path
         self.seed = seed
         self.pointvals = []
+        self.catmod = None
+
+        try:
+            catmod = SourceFileLoader(
+                'catmod',
+                os.path.join(path, 'category.py')).load_module()
+            assert all([
+                hasattr(catmod, 'make'),
+                hasattr(catmod, 'points'),
+                type(catmod.points) is list,
+            ])
+            self.catmod = catmod
+            self.pointvals.extend(catmod.points)
```

Comment:
I kind of agree with Patrick on this one, and it's fairly easy to handle.  

It's not so much that puzzle handles all these different cases and chooses one, instead it handles all of them in a well defined order.